cmd/agent/app/agent.go

```diff
@@ -356,5 +376,83 @@ func obtainCredentialsFromMemberCluster(clusterRestConfig *restclient.Config, op
 		return nil, fmt.Errorf("failed to get serviceAccount secret for impersonation from cluster(%s), error: %v", opts.ClusterName, err)
 	}
 
+	// create a ClusterRole in cluster.
+	clusterRole := &rbacv1.ClusterRole{}
+	clusterRole.Name = names.GenerateRoleName(impersonationSA.Name)
+	clusterRole.Rules = clusterPolicyRules
+	if _, err = ensureClusterRoleExist(clusterKubeClient, clusterRole); err != nil {
+		return nil, err
+	}
+
+	// create a ClusterRoleBinding in cluster.
+	clusterRoleBinding := &rbacv1.ClusterRoleBinding{}
+	clusterRoleBinding.Name = clusterRole.Name
+	clusterRoleBinding.Subjects = buildRoleBindingSubjects(impersonationSA.Name, impersonationSA.Namespace)
+	clusterRoleBinding.RoleRef = buildClusterRoleReference(clusterRole.Name)
+	if _, err = ensureClusterRoleBindingExist(clusterKubeClient, clusterRoleBinding); err != nil {
+		return nil, err
+	}
+
```

Comment:
We have create ClusterRole `karmada-impersonator` and ClusterRoleBinding `karmada-impersonator` for pull mode cluster, perhaps we do not need to create those rbac resource.

```
# kubectl --kubeconfig /root/.kube/members.config --context member3 get clusterrole | grep karmada-impersonator
karmada-impersonator                                                   2022-04-01T08:02:54Z
# kubectl --kubeconfig /root/.kube/members.config --context member3 get clusterrolebinding | grep karmada-impersonator
karmada-impersonator  
```

Reply:
Oh, I didn't see where to create these rbac, and let me see again.

Reply:
@XiShanYongYe-Chang I still can't find the exact location that create rbac for Pull cluster.

Reply:
Look here: pkg/controllers/unifiedauth/unified_auth_controller.go

`unified-auth-controller` will create impersonator rbac for all cluster(include pull mode).

Reply:
Does this controller`(unified-auth-controller`) need to be registered in `agent` component? Or directly manipulate PULL mode clusters in `karmada-controller-manager` component?

Reply:
`unified-auth-controller` will create `work` with `clusterrole/clusterrolebinding` for all cluster:

https://github.com/karmada-io/karmada/blob/7a1e96ea45817918c8df88b0b6e2a2b089a34901/pkg/controllers/unifiedauth/unified_auth_controller.go#L122-L132

Reply:
Got it, thanks!

Reply:
Have removed it.